client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -912,7 +912,14 @@ private static Exception extractAndWrapCause(Exception exception) {
             return e;
         }
         if (exception instanceof SSLHandshakeException) {
-            SSLHandshakeException e = new SSLHandshakeException(exception.getMessage());
+            SSLHandshakeException e;
+            if (exception.getMessage().contains("SunCertPathBuilderException")) {
+                e = new SSLHandshakeException(
+                    exception.getMessage() + "\nSee https://opensearch.org/docs/latest/clients/java-rest-high-level/ for troubleshooting."
+                );
+            } else {
```

Comment:
Shouldn't we just return `exception` in this case?

Reply:
I was following the code that was there before I made the change.  For all of the other types of exceptions, `e` is returned instead of `exception`.

Reply:
I mean in the `else` case the original exception is a `SSLHandshakeException`, so why wrap it again?

Reply:
I think it has something to do with wrapping but I'm not sure

Reply:
I'm not sure but the entire method is all extracting the type of exception and wrapping it

Reply:
Shouldn't be a concern now since we are removing the if..else and logging troubleshooting instructions for any SSLHandshakeException.

Reply:
@dblock i think the reason for wrapping could be to enhance the stack trace "readability" if I have to guess. For example, in this [stacktrace](https://gist.github.com/setiah/ecd58925379bf711a207196fecc4d10d) it is easy to notice SSLHandshakeException is coming from where in client code and what is the cause.